docs/source/app-dev/bindings-java/codegen.rst

```diff
@@ -380,3 +390,99 @@ The Java code generated for this variant is:
     public Published(Long year, String publisher) { /* ... */ }
     public Record toValue() { /* ... */ }
   }
+
+Parameterized types
+~~~~~~~~~~~~~~~~~~~
+
+A :ref:`parameterized type <daml-ref-parameterized-types>` is where the actual type is specified at instantiation as *type parameters* and not as part of type definition.
+The Java Code Generator uses Java Generic types to represent the DAML parameterized type.
+
+Below is a DAML fragment defining the parameterized type `Attribute` for use by the `
+
+.. code-block:: daml
+  :caption: Com/Acme.daml
+
+  daml 1.2
+  module Com.Acme where
+
+  data Attribute a = Attribute
+      with v : a
+
+  data BookAttributes = BookAttributes with
+     pages : (Attribute Int)
+     authors : (Attribute [Text])
+     title : (Attribute Text)
+
+A file Java file is generated for the `Attribute` data type that defines the Java Generic class:
+
+Note: If the parameterized type is contained in a type where the *actual* type is specified (as in the `BookAttributes` type above), then the serialization
+and deserialization of the enclosing type provides the necessary methods for serialization and deserialization.
+
+.. code-block:: java
+  :caption: com/acme/Attribute.java
+  :emphasize-lines: 3,8,10
+
+  package com.acme;
+
+  public class Attribute<a> {
+    public final a value;
+
+    public Attribute(a value) { /* ... */  }
+
+    public Record toValue(Function<a, Value> toValuea) { /* ... */ }
+
+    public static <a> Attribute<a> fromValue(Value value$, Function<Value, a> fromValuea) { /* ... */ }
+  }
+
+**Serializing**
```

Comment:
This and below - if it's meant to be a title, can you use an rst title? The [docs README](https://github.com/digital-asset/daml/tree/master/docs#style-conventions) has the order of headings we conventionally use.